pkg/core/profile/file.go

```diff
@@ -0,0 +1,134 @@
+package profile
```

Comment:
apart from naming - does context as package causes some clashes/problems?

Reply:
putting as `package context` makes things tough hile working with go's native context package, wanted to avoid additional rename.